### PR TITLE
Add a workflow for running tests using Github Actions.

### PR DESCRIPTION
**Pull request by snu5mumr1k**
Always use hirsute (so we use newer versions of vim) and allow neovim to fail (neovim hangs on user interactions).

### CODE REVIEW

**Comment by snu5mumr1k** [comment]
This is likely to resolve #192, because the Hirsute version is currently 2:8.2.2434-1ubuntu1.

NOTE: I've pinned Hirsute, because they've changed some of the ways they handle errors between 2:8.2.0716 and 2:8.2.2434. If we want to test more versions in the past, I'll have to change some stuff in the #246 near json tests, but then this pr will also fix part of #145 as it's basically allowing us to install bleeding edge versions from testing repos.

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          - false
+        exclude:
+          - os: ubuntu-18.04
+            vim-from-testing: true
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Add hirsute repos
+        if: ${{ matrix.os == 'ubuntu-latest' && matrix.vim-from-testing }}
+        run: sudo apt-add-repository 'deb http://archive.ubuntu.com/ubuntu/ impish main universe'
```
Comment (dbarnett) [inline .github/workflows/run-tests.yml:27]: It says "hirsute" above but "impish" here. Is that intentional?

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          sudo apt update
+          sudo apt install vim vim-gtk xvfb
+      - name: Install neovim
+        if: ${{ matrix.vim-flavour == 'neovim' }}
+        run: |
+          sudo apt update
+          sudo apt install neovim python3-neovim xvfb
+      - name: Install vroom
+        run: |
+          wget https://github.com/google/vroom/releases/download/v0.14.0/vroom_0.14.0-1_all.deb
```
Comment (dbarnett) [inline .github/workflows/run-tests.yml:41]: Nit: Could we put these versions in an env var, like in https://github.com/google/vim-codefmt/blob/793d816524934e6553c76437120eea5df8e85a1e/.travis.yml#L22?

**Review by dbarnett — APPROVED** [review]
Nice!

Should we be deleting .travis.yml along with this?

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (snu5mumr1k) [reply .github/workflows/run-tests.yml:27]: I don't think the bleeding edge will be added to Github Actions as an out-of-the-box option, so I'll put a simpler 'testing' here.

The idea is that by only changing the distribution name one can bump the testing.

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (snu5mumr1k) [reply .github/workflows/run-tests.yml]: Done.

**Comment by snu5mumr1k** [comment]
I've removed .travis.yml

**Pull request merged**